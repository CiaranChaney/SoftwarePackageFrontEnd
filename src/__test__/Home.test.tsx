import {screen, render} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import HomePage from '../pages/Home';
import '@testing-library/jest-dom';

beforeEach(() => {
    jest.doMock("../components/NavBar.tsx", () => {
        return () => (
            <div>
                <nav>
                    <ul>
                        <li>NavBar</li>
                    </ul>
                </nav>
            </div>
        );
    });
    jest.doMock("../components/Footer.tsx", () => () => <div>Footer</div>);
});


describe('HomePage', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <BrowserRouter>
                <HomePage />
            </BrowserRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('includes necessary subcomponents', () => {
        const { getByText } = render(
            <BrowserRouter>
                <HomePage />
            </BrowserRouter>
        );
        expect(document.querySelector('link[rel="icon"]')).toBeInTheDocument();
        expect(getByText('Library Hash Repository')).toBeInTheDocument();
        expect(getByText('Libraries')).toBeInTheDocument();
        expect(getByText('Login')).toBeInTheDocument();
        const registerButtons = screen.getAllByText('Register');
        registerButtons.forEach(button => {
            expect(button).toBeInTheDocument();
        });
        expect(getByText('Contact Us')).toBeInTheDocument();
        expect(getByText('Terms of Service')).toBeInTheDocument();
        expect(getByText('Privacy Policy')).toBeInTheDocument();
    });

    it('should render text', () => {
        const { getByText } = render(
            <BrowserRouter>
                <HomePage />
            </BrowserRouter>
        );
        expect(getByText('Welcome to the Library Data Repository')).toBeInTheDocument();
        expect(getByText('Recent Malicious Libraries Found')).toBeInTheDocument();
        expect(getByText('About Us')).toBeInTheDocument();
        expect(getByText('Get Started')).toBeInTheDocument();
    });


});
