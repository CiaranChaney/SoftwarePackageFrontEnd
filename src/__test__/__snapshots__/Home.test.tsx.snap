// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="navCol col-lg-6 mb-3"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <a
                class="link-light"
                href="/"
              >
                <h4>
                  Library Hash Repository
                </h4>
              </a>
            </div>
          </div>
        </div>
        <div
          class="navCol col-lg-6 d-flex justify-content-end mb-3"
        >
          <div
            class="row"
          >
            <div
              class="col-9 col-lg-4 col-sm-6 d-flex justify-content-end mb-2 mb-lg-0"
            >
              <a
                class="btn btn-light btn-lg p-2"
                href="/library"
                role="button"
              >
                Libraries
              </a>
            </div>
            <div
              class="col-3 col-lg-3 col-sm-6 d-flex justify-content-end mb-2 mb-lg-0"
            >
              <a
                class="btn btn-light btn-lg p-2"
                href="/login"
                role="button"
              >
                Login
              </a>
            </div>
            <div
              class="col-4 col-lg-4 col-sm-6 d-flex justify-content-end mb-2 mb-lg-0"
            >
              <a
                class="btn btn-light btn-lg p-2"
                href="/register"
                role="button"
              >
                Register
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <header
      class="text-center py-5"
    >
      <h1>
        Welcome to the Hash Veritas Repository
      </h1>
      <p
        class="lead"
      >
        Your go-to platform for managing and validating library data.
      </p>
    </header>
    <section
      class="py-5 bg-light"
      id="featured"
    >
      <div
        class="container"
      >
        <h2
          class="text-center mb-4"
        >
          Recent Malicious Libraries Found
        </h2>
        <div
          class="row"
        >
          <div
            class="col-md-4 mb-3"
          >
            <div
              class="card border-warning"
            >
              <div
                class="card-header bg-warning text-white"
              >
                Severe Vulnerability Alert
              </div>
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  io.trino:trino-client
                </h5>
                <h6
                  class="text-muted"
                >
                  Version 439 - Trino
                </h6>
                <p
                  class="fw-bold"
                >
                  CVSS Score: 7.4
                </p>
                <p
                  class="card-text"
                >
                  Deserialization of untrusted data can occur 
                </p>
                <p
                  class="fw-bold"
                >
                  Vulnerable Hash:
                </p>
                <p
                  class="text-monospace"
                >
                  c7bb819bd87639d0b5de393b02e31a58e36e3cec196928e1c28dfef08f64d5cb
                </p>
                <p
                  class="fw-bold"
                >
                  Fix version: 440
                </p>
                <p
                  class="fw-bold"
                >
                  Fix Hash:
                </p>
                <p
                  class="text-monospace"
                >
                  55e1eefe8a7524079c64d28cb781d71fd2f2fe69e827e288fb9b7461c22c7c95
                </p>
              </div>
              <div
                class="card-footer"
              >
                <small
                  class="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
          <div
            class="col-md-4 mb-3"
          >
            <div
              class="card"
            >
              <div
                class="card border-danger"
              >
                <div
                  class="card-header bg-danger text-white"
                >
                  Critical Vulnerability Alert
                </div>
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    org.ciaranchaney:libraryhashing-be
                  </h5>
                  <h6
                    class="text-muted"
                  >
                    Version 1.5.6 - Hash Veritas
                  </h6>
                  <p
                    class="fw-bold"
                  >
                    CVSS Score: 9.5
                  </p>
                  <p
                    class="card-text"
                  >
                    Malicious back door added in version 1.5.6
                  </p>
                  <p
                    class="fw-bold"
                  >
                    Vulnerable Hash:
                  </p>
                  <p
                    class="text-monospace"
                  >
                    6d586d5a13e0a4f9d20cf25eed2f26a5adf3fda1aad7a0c599ebde14890fce53
                  </p>
                  <p
                    class="fw-bold"
                  >
                    Mitigated version: 1.5.7
                  </p>
                  <p
                    class="fw-bold"
                  >
                    Mitigated Hash:
                  </p>
                  <p
                    class="text-monospace"
                  >
                    4b92abc6d4903e0859ba30b05d6483c8327a299186774c31581db46b05f708d4
                  </p>
                </div>
                <div
                  class="card-footer"
                >
                  <small
                    class="text-muted"
                  >
                    Last updated 5 hours ago
                  </small>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-md-4 mb-3"
          >
            <div
              class="card"
            >
              <div
                class="card border-warning"
              >
                <div
                  class="card-header bg-warning text-white"
                >
                  Severe Vulnerability Alert
                </div>
                <div
                  class="card-body"
                >
                  <h5
                    class="card-title"
                  >
                    io.github.yangziwen:quick-state
                  </h5>
                  <h6
                    class="text-muted"
                  >
                    Version 0.0.4 - yangziwen
                  </h6>
                  <p
                    class="fw-bold"
                  >
                    CVSS Score: 5.2
                  </p>
                  <p
                    class="card-text"
                  >
                    Out of date hashing algorithm used
                  </p>
                  <p
                    class="fw-bold"
                  >
                    Vulnerable Hash:
                  </p>
                  <p
                    class="text-monospace"
                  >
                    30f5105afdc17a5b78287e5f42df397f55f636f2bcf26c6ce1c089925c145161
                  </p>
                  <p
                    class="fw-bold"
                  >
                    Mitigated version: 0.0.5
                  </p>
                  <p
                    class="fw-bold"
                  >
                    Mitigated Hash:
                  </p>
                  <p
                    class="text-monospace"
                  >
                    b8b510d95518c673aeb78549a279050300134aad80c99dbc24fd93b52b13a4ee
                  </p>
                </div>
                <div
                  class="card-footer"
                >
                  <small
                    class="text-muted"
                  >
                    Last updated 1 day ago
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="py-5"
      id="cta"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-lg-8 mx-auto text-center"
          >
            <h2
              class="mb-4"
            >
              Get Started with Our Services
            </h2>
            <p
              class="lead mb-5"
            >
              Access our comprehensive repository or sign up to stay informed about the latest updates and security notices.
            </p>
            <div
              aria-label="Call to Action Buttons"
              class="btn-group"
              role="group"
            >
              <a
                class="btn btn-dark btn-lg px-4 me-2"
                href="/library"
                role="button"
              >
                <i
                  class="bi bi-archive"
                />
                 View Libraries
              </a>
              <a
                class="btn btn-dark btn-lg px-4"
                href="/register"
                role="button"
              >
                <i
                  class="bi bi-pencil-square"
                />
                 Register Now
              </a>
            </div>
          </div>
        </div>
      </div>
    </section>
    <footer
      class="bg-dark text-white text-center py-4"
    >
      <div
        class="container"
      >
        <nav>
          <ul
            class="list-inline mb-0"
          >
            <li
              class="list-inline-item"
            >
              <a
                class="text-white"
                href="/contact"
              >
                Contact Us
              </a>
            </li>
            <li
              class="list-inline-item"
            >
              <a
                class="text-white"
                href="/terms"
              >
                Terms of Service
              </a>
            </li>
            <li
              class="list-inline-item"
            >
              <a
                class="text-white"
                href="/privacy"
              >
                Privacy Policy
              </a>
            </li>
          </ul>
        </nav>
        <p
          class="mb-0"
        >
          © 2024 Library Data Repository. All rights reserved.
        </p>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
